tests/unit/modules/test_vsphere.py

```diff
@@ -1918,3 +1948,728 @@ def test_vcenter_proxy_return(self):
+        'urn:vmomi:InventoryServiceCategory:'
+        '2db5b00b-f211-4bba-ba42-e2658ebbb283:GLOBAL',
+        'urn:vmomi:InventoryServiceCategory:'
+        'cd847c3c-687c-4bd9-8e5a-0eb536f0a01d:GLOBAL',
+        'urn:vmomi:InventoryServiceCategory:'
+        'd51c24f9-cffb-4ce0-af56-7f18b6e649af:GLOBAL'
+    ]
+
+    list_tags_return = [
+        'urn:vmomi:InventoryServiceTag:'
+        'a584a83b-3015-45ad-8057-a3630613052f:GLOBAL',
+        'urn:vmomi:InventoryServiceTag:'
+        'db08019c-15de-4bbf-be46-d81aaf8d25c0:GLOBAL',
+        'urn:vmomi:InventoryServiceTag:'
+        'b55ecc77-f4a5-49f8-ab52-38865467cfbe:GLOBAL',
+        'urn:vmomi:InventoryServiceTag:'
+        'f009ab1b-e1b5-4c40-b8f7-951d9d716b39:GLOBAL',
+        'urn:vmomi:InventoryServiceTag:'
+        '102bb4c5-9b76-4d6c-882a-76a91ee3edcc:GLOBAL',
+        'urn:vmomi:InventoryServiceTag:'
+        'bb0350b4-85db-46b0-a726-e7c5989fc857:GLOBAL',
+        'urn:vmomi:InventoryServiceTag:'
+        '71d30f2d-bb23-48e1-995f-630adfb0dc89:GLOBAL'
+    ]
+
+    list_attached_tags_return = [
+        'urn:vmomi:InventoryServiceTag:'
+        'b55ecc77-f4a5-49f8-ab52-38865467cfbe:GLOBAL',
+        'urn:vmomi:InventoryServiceTag:'
+        'bb0350b4-85db-46b0-a726-e7c5989fc857:GLOBAL'
+    ]
+
+    list_create_category_return = [
+        'urn:vmomi:InventoryServiceCategory:'
+        '0af54c2d-e8cd-4248-931e-2f5807d8c477:GLOBAL'
+    ]
+
+    list_create_tag_return = [
+        'urn:vmomi:InventoryServiceCategory:'
+        '0af54c2d-e8cd-4248-931e-2f5807d8c477:GLOBAL'
+    ]
+
+    attach_tags_return = {
+        'Tag attached':
+            'urn:vmomi:InventoryServiceTag:'
+            'bb0350b4-85db-46b0-a726-e7c5989fc857:GLOBAL'
+    }
+
+    def test_create_tag_category_client_none(self):
+        get_details = MagicMock(return_value=self.details)
+
+        # Start patching each external API return with Mock Objects
+        with patch.object(vsphere,
+                          'get_proxy_type',
+                          return_value='vcenter'
+                          ) as get_proxy_type:
+            with patch.object(vsphere,
+                              '_get_proxy_connection_details',
+                              return_value=[]
+                              ) as get_proxy_connection:
+                with patch.object(salt.utils.vmware,
+                                  'get_service_instance',
+                                  return_value=None
+                                  ) as get_service_instance:
+                    with patch.dict(vsphere.__salt__,
+                                    {'vcenter.get_details': get_details},
+                                    clear=True) as get_vcenter_details:
+                        with patch.object(salt.utils.vmware,
+                                          'get_vsphere_client',
+                                          return_value=None
+                                          ) as get_vsphere_client:
```

Comment:
FWIW you can write this like:

```
patch_proxy_type = patch.object(vsphere, 'get_proxy_type', ...)
patch_proxy_connection = patch.object(...)

with patch_proxy_type as get_proxy_type:
    with patch_proxy_connection as get_proxy_connection:
        ...
```

I think you can also write it as

```
with patch_proxy_type as get_proxy_type, \
        patch_proxy_connection as get_proxy_connection, \
        ...:
    ...
```

If that feels cleaner

Reply:
I may, simply because these lines are SO long and I would prefer a cleaner way of writing it!